srtcore/core.cpp

```diff
@@ -8695,16 +8695,20 @@ void srt::CUDT::processCtrlAckAck(const CPacket& ctrlpkt, const time_point& tsAr
     // srt_recvfile (which doesn't make any sense), you'll have a deadlock.
     if (m_config.bDriftTracer)
     {
-        const bool drift_updated SRT_ATR_UNUSED = m_pRcvBuffer->addRcvTsbPdDriftSample(ctrlpkt.getMsgTimeStamp(), tsArrival, rtt);
 #if ENABLE_BONDING
-        if (drift_updated && m_parent->m_GroupOf)
-        {
-            ScopedLock glock(uglobal().m_GlobControlLock);
-            if (m_parent->m_GroupOf)
-            {
-                m_parent->m_GroupOf->synchronizeDrift(this);
-            }
-        }
+		if (m_parent->m_GroupOf)
+		{
+			ScopedLock glock(uglobal().m_GlobControlLock);
+			const bool drift_updated SRT_ATR_UNUSED = m_pRcvBuffer->addRcvTsbPdDriftSample(ctrlpkt.getMsgTimeStamp(), tsArrival, rtt);
+			if (drift_updated)
+				m_parent->m_GroupOf->synchronizeDrift(this); 
+		}
+		else
+		{
+#endif
+			m_pRcvBuffer->addRcvTsbPdDriftSample(ctrlpkt.getMsgTimeStamp(), tsArrival, rtt);
+#if ENABLE_BONDING
+		}
```

Comment:
Fix tabulation.

Reply:
Is that known that `m_parent->m_GroupOf` cannot be reset to NULL in the meantime of calling lock on GlobControlLock?

Reply:
Oh, right. Should be re-fixed.

Reply:
This way then.
```c++
    if (m_config.bDriftTracer)
    {
#if ENABLE_BONDING
        ScopedLock glock(uglobal().m_GlobControlLock);
        const bool drift_updated SRT_ATR_UNUSED =
            m_pRcvBuffer->addRcvTsbPdDriftSample(ctrlpkt.getMsgTimeStamp(), tsArrival, rtt);
    
        if (drift_updated)
            m_parent->m_GroupOf->synchronizeDrift(this);
#else
        m_pRcvBuffer->addRcvTsbPdDriftSample(ctrlpkt.getMsgTimeStamp(), tsArrival, rtt);
#endif
    }
```